src/stock-on-hand-mobile/pages/product-details.jsx

```diff
@@ -49,16 +91,131 @@ const ProductDetails = () => {
+            Header: () => 
+                    <div>
+                        Quantity <i className='fa fa-question-circle'/>
+                    </div>,
+            id: 'quantity',
+            accessor: 'quantity',
+            Cell: ({row}) => 
+                    <div className='stock-on-hand'>
+                        <div>
+                            {row.original.quantity}
+                        </div>
+                        <div>
+                            {`(${row.original.stockOnHand})`}
+                        </div>
+                    </div>
+           },
+            {
+                Header: ' ',
+                accessor: 'id',
+                Cell: ({row}) =>
+                    <i
+                        className={`fa fa-chevron-${isProductExpanded(getExpandedProducts(), row.original.id) ? 'up' : 'down'}`}
+                        aria-hidden='true'
+                        onClick={() => {
+                            handleExpandView(row.original.id);
+                        }}
+                    />
+            }],
+        []);
 
     const handleGoBack = () => {
         history.goBack();
     };
 
+    const getExpandedProducts = () => {
+        return JSON.parse(localStorage.getItem('expandedProductsBinCard')) || [];
+    };
+
+    const isProductExpanded = (expandedProducts, productId)=>  {
+        return expandedProducts.filter((expandedProductId) => expandedProductId == productId).length > 0;
+    };
+
+    const handleExpandView = (productId) => {
+        const expandedProducts = getExpandedProducts();
+        const isExpanded = isProductExpanded(expandedProducts, productId);
+
+        const expandedProductsToSet = isExpanded ?
+        expandedProducts.filter((expandedProductId) => expandedProductId != productId) :
+            [...expandedProducts, productId];
+
+        setExpandProductClicked(`${productId}${isExpanded}`);
+        localStorage.setItem('expandedProductsBinCard', JSON.stringify(expandedProductsToSet));
+    };
+
+    const getSignedQuantity = (adjustment) => {
+        return adjustment.reason.reasonType === REASON_TYPES.DEBIT ? -adjustment.quantity
+        : adjustment.quantity;
+    }
+
+    const getReason = (lineItem) => {
+        if (lineItem.reasonFreeText) {
+            return messageService.get('stockCard.reasonAndFreeText', {
+                name: lineItem.reason.name,
+                freeText: lineItem.reasonFreeText
+            });
+        }
+        return lineItem.reason.isPhysicalReason()
+            ? messageService.get('stockCard.physicalInventory')
+            : lineItem.reason.name;
+    };
+
+    const prepareStockCard = (stockCard) => {
+        const items = [];
+        let previousSoh;
+
+        stockCard.lineItems.forEach((lineItem) => {
+            if (lineItem.stockAdjustments.length > 0) {
+                lineItem.stockAdjustments.slice().forEach(function(adjustment, i) {
+                    let lineValue = ineItem.copy();
```

Comment:
is this typo? Seems to be for me @dariapionk 
should be here `let lineValue = lineItem.copy()`?